src/overlay/FlowControl.cpp

```diff
@@ -199,6 +189,42 @@ FlowControl::getNextBatchToSend()
     return batchToSend;
 }
 
+void
+FlowControl::updateMsgMetrics(std::shared_ptr<StellarMessage const> msg,
+                              VirtualClock::time_point const& timePlaced)
+{
```

Comment:
Why do we need to acquire the lock here? The metrics class is thread safe, as well as the medida metric classes themselves. is `mAppConnector.now()` not thread safe?

Reply:
it's not strictly needed (right now), i just put it to be consistent with the rest of the flow control class and avoid foot guns in the future in case this function gets modified.

Reply:
Cool, could you add a comment clarifying this? I was a little confused at first.

Reply:
Done!